src/Infrastructure/Identity/RoleService.cs

```diff
@@ -183,13 +194,17 @@ public async Task<string> UpdatePermissionsAsync(string roleId, List<UpdatePermi
             }
         }
 
-        var claims = await _roleManager.GetClaimsAsync(role);
-        foreach (var claim in claims.Where(c => permissions.Any(p => p.Permission == c.Value)))
+        var currentPermissions = await _roleManager.GetClaimsAsync(role);
+
+        // Remove permissions that were previously enabled
+        foreach (var claim in currentPermissions.Where(c => !selectedPermissions.Any(p => p.Permission == c.Value)))
         {
             await _roleManager.RemoveClaimAsync(role, claim);
         }
 
-        foreach (var permission in selectedPermissions)
+
+        // Add all permissions that were not previously enabled
+        foreach (var permission in selectedPermissions.Where(c => !currentPermissions.Any(p => p.Value == c.Permission)))
```

Comment:
Why this where clause? all currentpermissions have just been removed so unless I'm missing something, this condition will always return true... might as wel just do "foreach (var permission in selectedPermissions)", right?

Reply:
also the comment (and the previous one) needs updating... makes not much sense talking about "enabled" anymore...

Reply:
Hi, in the previous code, the remove claims one, we dont remove all claims, we only remove the ones that were previously selected, and are not anymore, and in this code, we only add the new selected ones, that´s why the where clauses

Reply:
Ok, got it now!